### PR TITLE
Remove support for Python 3.7

### LINKED ISSUE #426: Drop Python 3.7
Python 3.7 has a lot of typing baggage and is now no longer supported upstream, let's follow pytest and drop support for it.

### PR DESCRIPTION
**Pull request by bluetech**
Fix #426.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist=linting,docs,py{37,38,39,310,311,py3},py{38}-pytest{main}
+envlist=docs,py{38,39,310,311,py3},py{38}-pytest{main}
```
Comment (RonnyPfannschmidt) [inline tox.ini:2]: i presume linting was removed for pre-commit - it might be nice to have a changelog entry

**Review by RonnyPfannschmidt — APPROVED** [review]
lovely, thanks!

**Review by nicoddemus — APPROVED** [review]

File: tox.ini (hunk 1 above)
Comment (bluetech) [reply tox.ini:2]: Yes it was removed in d128bc5872e2c2ec16da5abdf2c0163267985b4f.

**Comment by bluetech** [comment]
Needed to add a `.readthedocs.yml` file to fix the RTD build which uses Python 3.7 by default.

**Pull request merged**